src/main/java/org/opentripplanner/routing/core/FareRuleSet.java

```diff
@@ -33,9 +37,20 @@ public Set<P2<String>> getOriginDestinations() {
     return originDestinations;
   }
 
+  public void addRouteOriginDestination(String route, String origin, String destination) {
```

Comment:
Can you add some information about what these methods mean?

Reply:
This is used for special routes in HSL where some tickets can be used outside the HSL service area when travelling between some origin  and destination on a certain route. All three pieces of information are needed for determining whether a special route rule needs to be applied when computing fare information